### PR TITLE
fix(vm): update disk size to 0 vmReadCustom

### PR DESCRIPTION
**Pull request by Jayden-Lind**
### Contributor's Note

- [ ] I have added / updated documentation in `/docs` for any user-facing features or additions.
- [ ] I have added / updated templates in `/example` for any new or updated resources / data sources.
- [x] I have ran `make example` to verify that the change works as expected.

_Sorry for the hasty PR, but just needed to highlight an issue and raise a PR for awareness._

When importing a VM into your state file that has no disk size specified, you receive a crash, which is shown down below.

```
[22:12:45][xxxxx@xxxxxDESKTOP ~/.../xxxxx-Terraform/proxmox](main *%)$ terraform import proxmox_virtual_environment_vm.xxxxx-plex-01 xxxxx-proxmox-01/107 
proxmox_virtual_environment_vm.xxxxx-plex-01: Importing from ID "xxxxx-proxmox-01/107"...
proxmox_virtual_environment_vm.xxxxx-plex-01: Import prepared!
  Prepared proxmox_virtual_environment_vm for import
proxmox_virtual_environment_vm.xxxxx-plex-01: Refreshing state... [id=107]
╷
│ Error: Plugin did not respond
│ 
│ The plugin encountered an error, and failed to respond to the plugin6.(*GRPCProvider).ReadResource call. The plugin logs may contain more details.
╵


Stack trace from the terraform-provider-proxmox_v0.46.1 plugin:

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xccbe8c]

goroutine 97 [running]:
github.com/bpg/terraform-provider-proxmox/proxmoxtf/resource.vmReadCustom({0x1086b10, 0xc000238a10}, 0xc00063aa80, {0xe4c8a0?, 0xc0004f77a0?}, 0x496745?, 0xc000265200, 0xc0002851c8?)
        github.com/bpg/terraform-provider-proxmox/proxmoxtf/resource/vm.go:4220 +0x39ec
github.com/bpg/terraform-provider-proxmox/proxmoxtf/resource.vmRead({0x1086b10, 0xc000238a10}, 0xc00063aa80, {0xe4c8a0?, 0xc0004f77a0?})
        github.com/bpg/terraform-provider-proxmox/proxmoxtf/resource/vm.go:3869 +0x791
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).read(0xc00033d340, {0x1086a68, 0xc000353230}, 0xd?, {0xe4c8a0, 0xc0004f77a0})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.31.0/helper/schema/resource.go:795 +0x11b
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).RefreshWithoutUpgrade(0xc00033d340, {0x1086a68, 0xc000353230}, 0xc000599d40, {0xe4c8a0, 0xc0004f77a0})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.31.0/helper/schema/resource.go:1089 +0x552
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ReadResource(0xc0001219c8, {0x1086a68?, 0xc0003530e0?}, 0xc0004e0400)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.31.0/helper/schema/grpc_provider.go:663 +0x48a
github.com/hashicorp/terraform-plugin-mux/tf5to6server.v5tov6Server.ReadResource({{0x108e200?, 0xc0001219c8?}}, {0x1086a68?, 0xc0003530e0?}, 0xc0004e0280?)
        github.com/hashicorp/terraform-plugin-mux@v0.13.0/tf5to6server/tf5to6server.go:176 +0x225
github.com/hashicorp/terraform-plugin-mux/tf6muxserver.(*muxServer).ReadResource(0x1086aa0?, {0x1086a68?, 0xc000352de0?}, 0xc0004e0280)
        github.com/hashicorp/terraform-plugin-mux@v0.13.0/tf6muxserver/mux_server_ReadResource.go:35 +0x193
github.com/hashicorp/terraform-plugin-go/tfprotov6/tf6server.(*server).ReadResource(0xc0002b5900, {0x1086a68?, 0xc0003523c0?}, 0xc0004222a0)
        github.com/hashicorp/terraform-plugin-go@v0.21.0/tfprotov6/tf6server/server.go:775 +0x4c3
github.com/hashicorp/terraform-plugin-go/tfprotov6/internal/tfplugin6._Provider_ReadResource_Handler({0xed09a0?, 0xc0002b5900}, {0x1086a68, 0xc0003523c0}, 0xc00063a000, 0x0)
        github.com/hashicorp/terraform-plugin-go@v0.21.0/tfprotov6/internal/tfplugin6/tfplugin6_grpc.pb.go:482 +0x169
google.golang.org/grpc.(*Server).processUnaryRPC(0xc000197000, {0x1086a68, 0xc000352330}, {0x108c7a8, 0xc000686000}, 0xc00050c000, 0xc000353ec0, 0x16a7f48, 0x0)
        google.golang.org/grpc@v1.61.0/server.go:1385 +0xe03
google.golang.org/grpc.(*Server).handleStream(0xc000197000, {0x108c7a8, 0xc000686000}, 0xc00050c000)
        google.golang.org/grpc@v1.61.0/server.go:1796 +0xfec
google.golang.org/grpc.(*Server).serveStreams.func2.1()
        google.golang.org/grpc@v1.61.0/server.go:1029 +0x8b
created by google.golang.org/grpc.(*Server).serveStreams.func2 in goroutine 52
        google.golang.org/grpc@v1.61.0/server.go:1040 +0x135

Error: The terraform-provider-proxmox_v0.46.1 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
```

Example VM setup that produced the panic:

image

### Proof of Work

After this quick fix:

```
[22:15:36][xxxxx@xxxxxDESKTOP ~/.../xxxxx-Terraform/proxmox](main *%)$ terraform import proxmox_virtual_environment_vm.xxxxx-plex-01 xxxxx-proxmox-01/107 
proxmox_virtual_environment_vm.xxxxx-plex-01: Importing from ID "xxxxx-proxmox-01/107"...
proxmox_virtual_environment_vm.xxxxx-plex-01: Import prepared!
  Prepared proxmox_virtual_environment_vm for import
proxmox_virtual_environment_vm.xxxxx-plex-01: Refreshing state... [id=107]

Import successful!

The resources that were imported are shown above. These resources are now in
your Terraform state and will henceforth be managed by Terraform.
```

### Community Note

- Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
- Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Closes #0000 | Relates #0000

### CODE REVIEW

File: proxmoxtf/resource/vm.go [hunk 1]
```diff
@@ -4217,6 +4217,11 @@ func vmReadCustom(
 		}
 
 		disk[mkResourceVirtualEnvironmentVMDiskInterface] = di
+
+		if dd.Size == nil {
+			dd.Size = new(types.DiskSize)
+		}
```
Comment (bpg-dev) [inline proxmoxtf/resource/vm.go]: Thanks @Jayden-Lind, that's a good catch!

This change definitely fixes the issue for your use case. However, the func `getDiskInfo` that populates the `CustomStorageDevice` struct is used in a few more places. 
So I would suggest to move this `if ...` statement over there, put it somewhere between [these 2 lines](https://github.com/bpg/terraform-provider-proxmox/blob/main/proxmoxtf/resource/vm.go#L6409-L6410).

Comment (Jayden-Lind) [reply proxmoxtf/resource/vm.go]: Thanks @bpg, will check it out tonight and update it 😄

Comment (Jayden-Lind) [reply proxmoxtf/resource/vm.go]: Updated @bpg

**Review by bpg — DISMISSED** [review]
Awesome! Thanks for the fix @Jayden-Lind

LGTM! 🚀

**Review by bpg — APPROVED** [review]

**Comment by bpg** [comment]
@all-contributors please add @Jayden-Lind for code

**Pull request merged**